src/data/Data.js

```diff
@@ -128,7 +122,7 @@ Data.prototype = {
 	},
 	remove: function( owner, key ) {
 		var i, name, camel,
-			cache = this.cache( owner );
+			cache = owner[ this.expando ];
```

Comment:
I didn't observe any such failures, but an early return is a nice optimization that doesn't appear to have any adverse side effects